### PR TITLE
Avoid spurious crash reporting

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
PTAL @charlesmchen-signal

### CODE REVIEW

File: Signal/src/AppDelegate.m [hunk 1]
```diff
@@ -306,7 +306,8 @@ - (void)verifyDBKeysAvailableBeforeBackgroundLaunch
         [[UIApplication sharedApplication] scheduleLocalNotification:notification];
         [UIApplication.sharedApplication setApplicationIconBadgeNumber:1];
 
-        OWSFail(@"!isDatabasePasswordAccessible.");
+        [DDLog flushLog];
```
Comment (charlesmchen-signal) [inline Signal/src/AppDelegate.m]: Shouldn't we log something before we exit?

File: Signal/src/environment/SignalApp.m [hunk 2]
```diff
@@ -140,7 +140,7 @@ + (void)resetAppData
     [self clearAllNotifications];
 
     [DebugLogger.sharedLogger wipeLogs];
-    OWSFail(@"App data reset.");
```
Comment (charlesmchen-signal) [inline Signal/src/environment/SignalApp.m]: Ditto.

Comment (charlesmchen-signal) [reply Signal/src/environment/SignalApp.m]: Well, maybe not here.

File: Signal/src/AppDelegate.m (hunk 1 above)
Comment (michaelkirk-signal) [reply Signal/src/AppDelegate.m]: Existing logging is up a couple lines, we just want to make sure it (and anything else) flushes before exiting.

File: Signal/src/environment/SignalApp.m (hunk 2 above)
Comment (michaelkirk-signal) [reply Signal/src/environment/SignalApp.m]: see line 133 above

**Pull request closed without merging**